apps/wallet/frontend/src/components/ListAllocationRequests.tsx

```diff
@@ -211,34 +225,65 @@ const TransferLegsDisplay: React.FC<{
 };
 
 const AllocationRequestActionButton: React.FC<{
+  id: string;
   allocationRequest: Contract<AllocationRequest>;
   allocations: Contract<AmuletAllocation>[];
   userParty: string;
   transferLegId: string;
-}> = () => {
-  /*TODO (#1100): uncomment and implement callback*/
-  return null;
-  // const transferLeg = allocationRequest.payload.transferLegs[transferLegId];
-  // const actionAllowed =
-  //   transferLeg.sender === userParty && transferLeg.instrumentId.id === 'Amulet';
-  // const alreadyAccepted = !!allocations.find(alloc =>
-  //   isAllocationForTransferLeg(alloc, allocationRequest, transferLegId)
-  // );
-  //
-  // if (!actionAllowed) return null;
-  // else if (alreadyAccepted) {
+}> = ({ id, allocationRequest, transferLegId, userParty, allocations }) => {
+  const transferLeg = allocationRequest.payload.transferLegs[transferLegId];
+  const actionAllowed =
+    transferLeg.sender === userParty && transferLeg.instrumentId.id === 'Amulet';
+  const settlement = allocationRequest.payload.settlement;
+  const alreadyAccepted = !!allocations.find(alloc =>
+    isAllocationForTransferLeg(alloc, allocationRequest, transferLegId)
+  );
+
+  const { createAllocation } = useWalletClient();
+  const createAllocationMutation = useMutation({
+    mutationFn: async () => {
+      const payload: AllocateAmuletRequest = openApiRequestFromTransferLeg(
+        settlement,
+        transferLeg,
+        transferLegId
+      );
+      return await createAllocation(payload);
+    },
+    onSuccess: () => {},
+    onError: error => {
+      console.error('Failed to submit allocation', error);
+    },
+  });
+
+  if (!actionAllowed) return null;
+  // TODO (#1413): show the withdraw button and implement the callback, instead of showing nothing
+  if (alreadyAccepted) return null;
```

Comment:
nit: move at the start of the function, doing a lot of stuff only to then return null seems pointless

Reply:
ah ofc forgot about the react hook conditions